### ISSUE TITLE
UFcalculator energies and energy calculated directly from features don't match

### ISSUE THREAD

**Issue by monk-04** [uf3/uf3#69]
If we calculate energies (and forces) directly from features as shown below-

```
feature_handler = process.BasisFeaturizer(bspline_config=model.bspline_config)

f2b = feature_handler.featurize_energy_2B(geom)
f3b = feature_handler.featurize_energy_3B(geom)
features_2b_3b = np.concatenate(([0],[0],f2b,f3b))
print("Energy from flat:",np.matmul(features_2b_3b,model.coefficients))
```

and using UFcalculator as shown below-

```
calc = calculator.UFCalculator(model)
geom.set_calculator(calc)
coeffs = model.dump()["coefficients"]

one_body_energy = 0
for i in set(geom.get_chemical_symbols()):
    one_body_energy += coeffs[i]*geom.get_chemical_symbols().count(i)
    
print("Energy:", geom.get_potential_energy()-(one_body_energy))
```

The two energies don't match. Similarly, if we calculate the forces they don't match.
(Here model is some learned model)

One possible reason for this might be that when featurize using `featurize_energy_3B` we are not double counting the 3B triangles, while in UFCalculator we are double counting the 3B triangles.

**Comment by monk-04** [comment]
It seems like this is an older bug and the latest changes to fix the symmetry bug didn't introduced this issue.

I am attaching a jupyter notebook ([ref_data.zip](https://github.com/uf3/uf3/files/12600391/ref_data.zip)) and a model json that clearly shows this issue. The jupyter notebook should be run using the UF3 version from 24th of August 2023 ([here](https://github.com/uf3/uf3/tree/3e5c7929f29de374e3c19fc1e3dc26e4c5ccacc1)).


 (**BEWARE** The model json file should not be used for any MD simulation, it is only for testing purpose)

Interestingly the energies calculated using the features and UFCalculator match for molecules in a box.

They don't match for periodic crystal geometry.

**Comment by sunghjung3** [comment]
There shouldn't be a bug when UF3 featurizes. The issue is how the Jupyter notebook calls the featurization methods:

    f2b = feature_handler.featurize_energy_2B(geom)
    f3b = feature_handler.featurize_energy_3B(geom)
    ...
    f2bf = feature_handler.featurize_force_2B(geom)
    f3bf = feature_handler.featurize_force_3B(geom)

These methods take in an optional argument called `supercell`, which should be created beforehand to include all relevant ghost atoms in neighboring cells. If `supercell` is not passed in, it defaults to using `geom` as the supercell. It makes sense why the difference in the calculated values was observed only for bulk but not for a molecule.

After creating a supercell like this and passing it in to the featurization methods, the energy values agree for the bulk structure:

    from uf3.data import geometry
    r_cut = feature_handler.r_cut
    supercell_Nb6Sn2 = geometry.get_supercell(geom_Nb6Sn2, r_cut=r_cut)

    f2b = feature_handler.featurize_energy_2B(geom, supercell=supercell_Nb6Sn2)
    ...

In UF3, this shouldn't be an issue because the supercell is handled by [`evaluate_configurations()`](https://github.com/uf3/uf3/blob/c8ac183d7e0d6ad06a88558eb384303f576089a2/uf3/representation/process.py#L332). A high-level flowchart of the featurization goes something like:

> 1. `geom`(s) is converted into `df_data` with `uf3.data.io.DataCoordinator.dataframe_from_lists()`
> 2. `df_data` is passed into `uf3.representation.process.BasisFeaturizer.evaluate_parallel()`
> 3. `evaluate_parallel()` --> `evaluate()` --> `evaluate_configuration()` (this is where the supercell is handled) --> `featurize_energy_2B()`, `featurize_force_2B()`, ...

**Comment by monk-04** [comment]
Yup you are correct @sunghjung3. I was not using the `featurize` method correctly, that's why I removed the bug label. In order to close the issue I will do two things-

1. Unit test to compare UFCalculator energy and forces with energy and forces obtained from representation.
2. Add warning when `featurize_energy_*` is called without a supercell as an input.
